File: flutter/lib/core/constants/api_constants.dart
class ApiConstants {
  static const String baseUrl = 'http://localhost:8080';
  static const String apiPrefix = '/api/v1';

  // Auth endpoints
  static const String register = '$apiPrefix/auth/register';
  static const String login = '$apiPrefix/auth/login';

  // Account endpoints
  static const String accounts = '$apiPrefix/accounts';

  // Category endpoints
  static const String categories = '$apiPrefix/categories';

  // Transaction endpoints
  static const String transactions = '$apiPrefix/transactions';

  // Sync endpoints
  static const String syncPull = '$apiPrefix/sync/pull';
  static const String syncPush = '$apiPrefix/sync/push';

  // WebSocket
  static const String wsSync = '/ws/sync';

  // Timeout
  static const int connectTimeout = 15000;
  static const int receiveTimeout = 15000;
}

File: web/src/core/constants/api.constants.ts
export const API_CONSTANTS = {
  baseUrl: 'http://localhost:8080/api/v1',
  apiPrefix: '/api/v1',
  wsBaseUrl: 'ws://localhost:8080',

  // Auth endpoints
  auth: {
    register: '/auth/register',
    login: '/auth/login',
  },

  // Account endpoints
  accounts: '/accounts',

  // Category endpoints
  categories: '/categories',

  // Transaction endpoints
  transactions: '/transactions',
  transactionsRange: '/transactions/range',
  transactionsStats: '/transactions/stats',
  transactionsStatsDetailed: '/transactions/stats/detailed',

  // Sync endpoints
  syncPull: '/sync/pull',
  syncPush: '/sync/push',

  // Import endpoints
  importSources: '/import/sources',
  importTemplate: '/import/template',
  importUpload: '/import/upload',
  importExecute: '/import/execute',

  // WebSocket
  wsSync: '/ws/sync',
} as const;
